cpu_features.h

```diff
@@ -34,14 +35,20 @@ extern uint32_t adler32_vmx(uint32_t adler, const unsigned char *buf, size_t len
 #ifdef X86_SSSE3_ADLER32
 extern uint32_t adler32_ssse3(uint32_t adler, const unsigned char *buf, size_t len);
 #endif
+#ifdef X86_SSE42_ADLER32
```

Comment:
Let's group all the `adler32_fold_copy_*` functions into their own section, similar to how `crc32` and `crc32_fold` functions are separated..